### PR TITLE
Fix multi-asic behaviour for pg-drop

### PR DESCRIPTION
**Pull request by bktsim-arista**
#### What I did
`show priority-group drop` is not behaving correctly on multi-asic devices, as the namespace option '-n' is not available and correct namespaces were not traversed to retrieve drop counters. This change fixes the multi-asic behaviour of this command and adds unit tests to ensure that the functionality is correct.

This is a part of the set of changes being pushed for https://github.com/sonic-net/sonic-buildimage/issues/15148

#### How I did it
Added multi-asic support and relevant unit tests for pg-drop script.

#### How to verify it
Run unit test, or `pg-drop -c show` with `-n`

### CODE REVIEW

**Comment by vmittal-msft** [comment]
@bktsim-arista Few things -
1. Can you please share how "Show priority-group drop/persistent-watermark/watermark" command is going to change ? 
2. please add test for all options under priority-group.
3. If no namespace specified, Will it show for all namespaces? if we specify, will it show for only that namespace ?
4. Is this verified to be working fine on pizza box as well as chassis?

**Review by vmittal-msft — COMMENTED** [review]
Please check comments

**Comment by arista-nwolfe** [comment]
> @bktsim-arista Few things -
> 
> 1. Can you please share how "Show priority-group drop/persistent-watermark/watermark" command is going to change ?
> 2. please add test for all options under priority-group.
> 3. If no namespace specified, Will it show for all namespaces? if we specify, will it show for only that namespace ?
> 4. Is this verified to be working fine on pizza box as well as chassis?

For questions 1 & 2: This PR only adds multi-asic support for `show priority-group drop` so we shouldn't have tests for `persistent-watermark/watermark` commands. Support for multi-asic on those commands along with their UT are in https://github.com/sonic-net/sonic-utilities/pull/3060

**Comment by kenneth-arista** [comment]
> 4\. Is this verified to be working fine on pizza box as well as chassis?

Yes, confirmed that the changes work on a pizza box as well.

**Comment by kenneth-arista** [comment]
> 3. If no namespace specified, Will it show for all namespaces? if we specify, will it show for only that namespace ?

If `--namespace` is specified, then only counters relevant to that namespace will be shown or cleared. If the namespace arg is not passed, the command affects all namespaces. 

For example,
```
$ show priority-group  drop counters --namespace asic0
Ingress PG dropped packets:
         Port    PG0    PG1    PG2    PG3    PG4    PG5    PG6    PG7
-------------  -----  -----  -----  -----  -----  -----  -----  -----
    Ethernet0      0      0      0      0      0      0      0      0
    Ethernet8      0      0      0      0      0      0      0      0
   Ethernet16      0      0      0      0      0      0      0      0
   Ethernet24      0      0      0      0      0      0      0      0
   Ethernet32      0      0      0      0      0      0      0      0
   Ethernet40      0      0      0      0      0      0      0      0
   Ethernet48      0      0      0      0      0      0      0      0
   Ethernet56      0      0      0      0      0      0      0      0
   Ethernet64      0      0      0      0      0      0      0      0
   Ethernet72      0      0      0      0      0      0      0      0
   Ethernet80      0      0      0      0      0      0      0      0
   Ethernet88      0      0      0      0      0      0      0      0
   Ethernet96      0      0      0      0      0      0      0      0
  Ethernet104      0      0      0      0      0      0      0      0
  Ethernet112      0      0      0      0      0      0      0      0
  Ethernet120      0      0      0      0      0      0      0      0
  Ethernet128      0      0      0      0      0      0      0      0
  Ethernet136      0      0      0      0      0      0      0      0
 Ethernet-IB0      0      0      0      0      0      0      0      0
Ethernet-Rec0      0      0      0      0      0      0      0      0
```
```
$ show priority-group  drop counters --namespace asic1
Ingress PG dropped packets:
         Port    PG0    PG1    PG2    PG3    PG4    PG5    PG6    PG7
-------------  -----  -----  -----  -----  -----  -----  -----  -----
  Ethernet144      0      0      0      0      0      0      0      0
  Ethernet152      0      0      0      0      0      0      0      0
  Ethernet160      0      0      0      0      0      0      0      0
  Ethernet168      0      0      0      0      0      0      0      0
  Ethernet176      0      0      0      0      0      0      0      0
  Ethernet184      0      0      0      0      0      0      0      0
  Ethernet192      0      0      0      0      0      0      0      0
  Ethernet200      0      0      0      0      0      0      0      0
  Ethernet208      0      0      0      0      0      0      0      0
  Ethernet216      0      0      0      0      0      0      0      0
  Ethernet224      0      0      0      0      0      0      0      0
  Ethernet232      0      0      0      0      0      0      0      0
  Ethernet240      0      0      0      0      0      0      0      0
  Ethernet248      0      0      0      0      0      0      0      0
  Ethernet256      0      0      0      0      0      0      0      0
  Ethernet264      0      0      0      0      0      0      0      0
  Ethernet272      0      0      0      0      0      0      0      0
  Ethernet280      0      0      0      0      0      0      0      0
 Ethernet-IB1      0      0      0      0      0      0      0      0
Ethernet-Rec1      0      0      0      0      0      0      0      0
```
```
$ show priority-group  drop counters
Ingress PG dropped packets:
         Port    PG0    PG1    PG2    PG3    PG4    PG5    PG6    PG7
-------------  -----  -----  -----  -----  -----  -----  -----  -----
    Ethernet0      0      0      0      0      0      0      0      0
    Ethernet8      0      0      0      0      0      0      0      0
   Ethernet16      0      0      0      0      0      0      0      0
   Ethernet24      0      0      0      0      0      0      0      0
   Ethernet32      0      0      0      0      0      0      0      0
   Ethernet40      0      0      0      0      0      0      0      0
   Ethernet48      0      0      0      0      0      0      0      0
   Ethernet56      0      0      0      0      0      0      0      0
   Ethernet64      0      0      0      0      0      0      0      0
   Ethernet72      0      0      0      0      0      0      0      0
   Ethernet80      0      0      0      0      0      0      0      0
   Ethernet88      0      0      0      0      0      0      0      0
   Ethernet96      0      0      0      0      0      0      0      0
  Ethernet104      0      0      0      0      0      0      0      0
  Ethernet112      0      0      0      0      0      0      0      0
  Ethernet120      0      0      0      0      0      0      0      0
  Ethernet128      0      0      0      0      0      0      0      0
  Ethernet136      0      0      0      0      0      0      0      0
  Ethernet144      0      0      0      0      0      0      0      0
  Ethernet152      0      0      0      0      0      0      0      0
  Ethernet160      0      0      0      0      0      0      0      0
  Ethernet168      0      0      0      0      0      0      0      0
  Ethernet176      0      0      0      0      0      0      0      0
  Ethernet184      0      0      0      0      0      0      0      0
  Ethernet192      0      0      0      0      0      0      0      0
  Ethernet200      0      0      0      0      0      0      0      0
  Ethernet208      0      0      0      0      0      0      0      0
  Ethernet216      0      0      0      0      0      0      0      0
  Ethernet224      0      0      0      0      0      0      0      0
  Ethernet232      0      0      0      0      0      0      0      0
  Ethernet240      0      0      0      0      0      0      0      0
  Ethernet248      0      0      0      0      0      0      0      0
  Ethernet256      0      0      0      0      0      0      0      0
  Ethernet264      0      0      0      0      0      0      0      0
  Ethernet272      0      0      0      0      0      0      0      0
  Ethernet280      0      0      0      0      0      0      0      0
 Ethernet-IB0      0      0      0      0      0      0      0      0
 Ethernet-IB1      0      0      0      0      0      0      0      0
Ethernet-Rec0      0      0      0      0      0      0      0      0
Ethernet-Rec1      0      0      0      0      0      0      0      0
```

**Comment by kenneth-arista** [comment]
For simplicity the `clear` command ignores the namespace argument and will always apply to all applicable namespaces. The reason is that clearing is accomplished via caching. If the namespace argument were honored, there will be unwieldy logic to cache only counters specific to the namespace. If the user mixes and matches `clear` and `show` commands with different `namespace` arguments, the end result will be confusing.

**Comment by kenneth-arista** [comment]
@arlakshm please review

File: scripts/pg-drop [hunk 1]
```diff
@@ -43,28 +48,26 @@ def get_dropstat_dir():
...
 
-    def __init__(self):
-        self.counters_db = SonicV2Connector(host='127.0.0.1')
-        self.counters_db.connect(self.counters_db.COUNTERS_DB)
-
-        self.configdb = ConfigDBConnector()
+    def __init__(self, namespace):
+        self.namespace = namespace
+        self.ns_list = multi_asic.get_namespace_list(namespace)
+        self.configdb = ConfigDBConnector(namespace=namespace, use_unix_socket_path=True)
```
Comment (kenneth-arista) [inline scripts/pg-drop:54]: Let's not use unix_sockets everywhere as this will require users with elevated privilege to run this show command.

**Review by wenyiz2021 — APPROVED** [review]

**Comment by saksarav-nokia** [comment]
@vmittal-msft , When will this get cherry-picked to 202405?

**Comment by vmittal-msft** [comment]
@kenneth-arista @bktsim-arista Can you please raise a PR for 2020405 ?

**Comment by rlhui** [comment]
@bingwang-ms  can we approve the back-port request? thanks.

**Pull request merged**